controllers/managedocs_controller.go

```diff
@@ -1067,6 +1085,52 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getPrometheusRemoteWriteSpec() (*promv1.RemoteWriteSpec, error) {
+	if err := r.get(r.rhobsSecret); err != nil {
+		return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+	}
+	rhobsSecretData := r.rhobsSecret.Data
+	rhobsUserID := rhobsSecretData[rhobsClientId]
+	if rhobsUserID == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", rhobsClientId)
+	}
+	rhobsUserAccessKey := rhobsSecretData[rhobsClientSecret]
+	if rhobsUserAccessKey == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", rhobsClientSecret)
+	}
```

Comment:
See last comment for reasoning:
```suggestion
	if _, found := rhobsSecretData["prom-remote-write-config-secret"]; !found {
		return nil, fmt.Error("rhobs secret does not contain a value for key prom-remote-write-config-secret")
	}
```